include-build/roborazzi-core/src/commonJvmMain/kotlin/com/github/takahirom/roborazzi/ImageIoFormat.commonJvm.kt

```diff
@@ -0,0 +1,123 @@
+import java.io.File
+import javax.imageio.IIOImage
+import javax.imageio.ImageIO
+import javax.imageio.ImageTypeSpecifier
+import javax.imageio.ImageWriteParam
+import javax.imageio.ImageWriter
+import javax.imageio.metadata.IIOMetadata
+import javax.imageio.metadata.IIOMetadataFormatImpl
+import javax.imageio.metadata.IIOMetadataNode
+import javax.imageio.stream.FileImageOutputStream
+
+@ExperimentalRoborazziApi
+@Suppress("FunctionName")
+actual fun LosslessWebPImageIoFormat(): ImageIoFormat {
+  return JvmImageIoFormat(
+    awtImageWriter = losslessWebPWriter()
+  )
+}
+
+@ExperimentalRoborazziApi
+actual fun ImageIoFormat(): ImageIoFormat {
+  return JvmImageIoFormat()
+}
+
+@ExperimentalRoborazziApi
+fun interface AwtImageWriter {
+  fun write(
+    destFile: File,
+    contextData: Map<String, Any>,
+    image: BufferedImage
+  )
+}
+
+@ExperimentalRoborazziApi
+fun interface AwtImageLoader {
+  fun load(inputFile: File): BufferedImage
+}
+
+@ExperimentalRoborazziApi
+data class JvmImageIoFormat(
+  val awtImageWriter: AwtImageWriter = AwtImageWriter { file, contextData, bufferedImage ->
+    val imageExtension = file.extension.ifBlank { "png" }
+    if (contextData.isEmpty()) {
+      ImageIO.write(
+        bufferedImage,
+        imageExtension,
+        file
+      )
+      return@AwtImageWriter
+    }
+    val writer = getWriter(bufferedImage, imageExtension)
+    val meta = writer.writeMetadata(contextData, bufferedImage)
+    writer.output = ImageIO.createImageOutputStream(file)
+    writer.write(IIOImage(bufferedImage, null, meta))
+  },
+  val awtImageLoader: AwtImageLoader = AwtImageLoader { ImageIO.read(it) }
+) : ImageIoFormat
+
+
+@ExperimentalRoborazziApi
+fun getWriter(renderedImage: RenderedImage, extension: String): ImageWriter {
+  val typeSpecifier = ImageTypeSpecifier.createFromRenderedImage(renderedImage)
+  val iterator: Iterator<*> = ImageIO.getImageWriters(typeSpecifier, extension)
+  return if (iterator.hasNext()) {
+    iterator.next() as ImageWriter
+  } else {
+    throw IllegalArgumentException("No ImageWriter found for $extension")
+  }
+}
+
+@ExperimentalRoborazziApi
+fun ImageWriter.writeMetadata(
+  contextData: Map<String, Any>,
+  bufferedImage: BufferedImage,
+): IIOMetadata? {
+  val meta = getDefaultImageMetadata(ImageTypeSpecifier(bufferedImage), null) ?: run {
+    // If we use WebP, it seems that we can't get the metadata
+    return null
+  }
+
+  val root = IIOMetadataNode(IIOMetadataFormatImpl.standardMetadataFormatName)
+  contextData.forEach { (key, value) ->
+    val textEntry = IIOMetadataNode("TextEntry")
+    textEntry.setAttribute("keyword", key)
+    textEntry.setAttribute("value", value.toString())
+    val text = IIOMetadataNode("Text")
+    text.appendChild(textEntry)
+    root.appendChild(text)
+  }
+
+  meta.mergeTree(IIOMetadataFormatImpl.standardMetadataFormatName, root)
+  return meta
+}
+
+/**
+ * Add testImplementation("io.github.darkxanter:webp-imageio:0.3.0") to use this
```

Comment:
every1 who wants to use webp format should add this?
is so, why not adding it as transitive dependency?

Reply:
Thanks, that makes sense. However, since we need to download additional code when using this, there are trade-offs to consider. Additionally, the roborazzi-core module is already being utilized by the Roborazzi Gradle Plugin. While we may need to add the roborazzi-webp module, it could complicate maintenance. This is why I'm using a compileOnly dependency. It would be beneficial to add a comment explaining the compileOnly dependency.
<img width="1262" alt="image" src="https://github.com/user-attachments/assets/e2a82c8c-4568-48ff-8e4a-3d8f9f6aed58">

Reply:
makes sense, yeah
should be enough for now to have it in readme